docs/src/operator/architecture.md

```diff
@@ -0,0 +1,47 @@
+# Node architecture
+
+The node itself consists of three distributed components: store, block-producer and RPC. We also provide a reference
+faucet implementation which we use to distribute testnet and devnet tokens.
+
+The components can be run on separate instances when optimised for performance, but can also be run as a single process
+for convenience. At the moment both of Miden's public networks (testnet and devnet) are operating in single process
+mode.
+
+The inter-component communication is done using a gRPC API wnich is assumed trusted. In other words this _must not_ be
+public. External communication is handled by the RPC component with a separate external-only gRPC API.
+
+![node architecture](../resources/operator_architecture.svg)
```

Comment:
I know this is my picture - but I think a couple of things here are confusing:

- The purpose of the explorer box is not super clear here. I think what we try to show is that "explorer.testnet.miden.io" gets redirected to it - but this is not really obvious.
- The "faucet" box is also a bit confusing - it feels like the faucet is not really communicating with any other component - but I think it does talk to an internal RPC, right?

Reply:
I've made some changes to the diagram:

- pass through arrows connecting the load balancer inputs to outputs
- marked the explorer as an external service
- added provers
- added missing links